simple_certmanager/admin.py

```diff
@@ -1,13 +1,89 @@
+from io import BytesIO
+from zipfile import ZipFile
+
 from django.contrib import admin
+from django.http import HttpResponse, HttpResponseRedirect
+from django.urls import reverse
 from django.utils.translation import gettext_lazy as _
 
 from privates.admin import PrivateMediaMixin
 
 from .forms import CertificateAdminForm
-from .models import Certificate
+from .models import Certificate, SigningRequest
 from .utils import suppress_cryptography_errors
 
 
+def download_csr(modeladmin, request, queryset):
+    if (num := len(queryset)) > 1:
+        zip_file = BytesIO()
+        with ZipFile(zip_file, "w") as zipf:
+            for csr in queryset:
+                csr_content = csr.csr
+                csr_filename = f"{csr.common_name}_csr.pem"
```

Comment:
1. I'd call `django.utils.text.slugify` on the common_name too, in the event that someone uses a character like `/` that messes with filenames
2. I suppose it's not impossible that multiple CSRs have the same common name, so to guarantee unique names, you can include the PK

```suggestion
                csr_filename = f"{slugify(csr.common_name)}_{csr.pk}_csr.pem"
```